src/pydvl/utils/score.py

```diff
@@ -81,6 +87,153 @@ def __repr__(self):
+        transformed_range = (0, disc_score_in_cls * disc_score_out_of_cls)
+        super().__init__(
+            "accuracy",
+            range=transformed_range,
+            default=default,
+            name=name or f"class wise {scoring}",
+        )
+        self._in_cls_discount_fn = in_class_discount_fn
+        self._out_of_cls_discount_fn = out_of_class_discount_fn
+        self.label = initial_label
+
+    def __str__(self):
+        return self._name
+
+    def __call__(
+        self: "ClassWiseScorer",
+        model: SupervisedModel,
+        x_test: NDArray[np.float_],
+        y_test: NDArray[np.int_],
+    ) -> float:
+        """
+        :param model: Model to be used for determining the score on the validation set.
+        :param x_test: Array containing the features of the associated classification
+            problem.
+        :param y_test: Array containing the labels of the associated classification
+            problem.
+        :return: The calculated score.
+        """
+        in_cls_score, out_of_cls_score = self.estimate_in_cls_and_out_of_cls_score(
+            model, x_test, y_test
+        )
+        disc_score_in_cls = self._in_cls_discount_fn(in_cls_score)
+        disc_score_out_of_cls = self._out_of_cls_discount_fn(out_of_cls_score)
+        return disc_score_in_cls * disc_score_out_of_cls
+
+    def estimate_in_cls_and_out_of_cls_score(
+        self,
+        model: SupervisedModel,
+        x_test: NDArray[np.float_],
+        y_test: NDArray[np.int_],
+        rescale_scores: bool = True,
+    ) -> Tuple[float, float]:
+        r"""
+        Computes in-cls and out-of-cls score by using the scoring function as can be
+        written as
+
+        .. math::
+            a_S(D=\{(\hat{x}_1, \hat{y}_1), \dots, (\hat{x}_K, \hat{y}_K)\}) &=
+            \frac{1}{N} \sum_k s(y(\hat{x}_k), \hat{y}_k) \\
+
+        whereas D_i contains only samples with label ``label`` from the validation set
+        and D_o contains all the other samples. By default, scores are scaled to be of
+        the same order of magnitude. In that case the raw scores are multiplied with
+
+        .. math::
+            N_{y_i} = \frac{a_S(D_{y_i})}{a_S(D_{y_i})+a_S(D_{-y_i})} \quad \text{and}
+            \quad N_{-y_i} = \frac{a_S(D_{-y_i})}{a_S(D_{y_i})+a_S(D_{-y_i})}
+
+        these factors to obtain the rescaled scores.
+
+        :param model: Model to be used for determining the score on the validation set.
+        :param x_test: Array containing the features of the associated classification
+            problem.
+        :param y_test: Array containing the labels of the associated classification
+            problem.
+        :param rescale_scores: True if the scores shall be denormalized. This makes
+            sense when for the inner score an estimator of the form 1/N sum_i x_i has
+            been used.
+        :return: Tuple containing the in-cls and out-of-cls score.
```

Comment:
Please review this docstring (and all others) for grammar, spelling and other errors. Also, explain all notation properly. Same for docs!